streamflow/deployment/filter/matching.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from collections.abc import MutableMapping, MutableSequence
+from importlib.resources import files
+
+from streamflow.core.deployment import BindingFilter, Target
+from streamflow.core.exception import (
+    WorkflowDefinitionException,
+    WorkflowExecutionException,
+)
+from streamflow.core.workflow import Job
+from streamflow.log_handler import logger
+from streamflow.workflow.token import FileToken, ListToken, ObjectToken
+
+
+class MatchingRule:
+    def __init__(
+        self,
+        deployment_name: str,
+        service: str | None,
+        predicates: MutableMapping[str, str],
+    ) -> None:
+        self.deployment: str = deployment_name
+        self.service: str | None = service
+        self.predicates: MutableMapping[str, str] = predicates
+
+    def eval(
+        self, filter_name: str, deployment_name: str, service: str | None, job: Job
```

Comment:
If each `MatchingRule` belongs to a single filter, just pass the `filter` to the constructor as an additional parameter